### ISSUE TITLE
Rai without audio

### ISSUE THREAD

**Issue by avagraha** [ytdl-org/youtube-dl#27830]
## Checklist

- [x ] I'm asking a question
- [ x] I've looked through the README and FAQ for similar questions
- [ x] I've searched the bugtracker for similar questions including closed ones

## Question

WRITE QUESTION HERE

1 of 3)

C:\Users\andrea>y https://www.raiplay.it/video/2021/01/Che-Dio-ci-aiuti-S6E6
-Deja-vu-383bca47-d8a1-49c0-876c-f20a2d56d4c1.html -F
[RaiPlay] 383bca47-d8a1-49c0-876c-f20a2d56d4c1: Downloading video JSON
[RaiPlay] 383bca47-d8a1-49c0-876c-f20a2d56d4c1: Downloading XML metadata fo
r platform mon
[RaiPlay] 383bca47-d8a1-49c0-876c-f20a2d56d4c1: Downloading m3u8 informatio
n
[RaiPlay] 383bca47-d8a1-49c0-876c-f20a2d56d4c1: Downloading XML metadata fo
r platform flash
[RaiPlay] 383bca47-d8a1-49c0-876c-f20a2d56d4c1: Downloading XML metadata fo
r platform native
[info] Available formats for 383bca47-d8a1-49c0-876c-f20a2d56d4c1:
format code               extension  resolution note
hls-aac-Audiodescrizione  mp4        audio only [Audiodescrizione]
hls-aac-Italiano          mp4        audio only [Italiano]
hls-1200                  mp4        704x396    1172k
hls-1800                  mp4        928x522    1758k
hls-2400                  mp4        1024x576   2100k  (best)

2 of 3)

C:\Users\andrea>y https://www.raiplay.it/video/2021/01/Che-Dio-ci-aiuti-S6E
-Obbligo-o-verita-027f044e-836d-4a3b-b1da-8eaa81715e3d.html
[RaiPlay] 027f044e-836d-4a3b-b1da-8eaa81715e3d: Downloading video JSON
[RaiPlay] 027f044e-836d-4a3b-b1da-8eaa81715e3d: Downloading XML metadata f
r platform mon
[RaiPlay] 027f044e-836d-4a3b-b1da-8eaa81715e3d: Downloading m3u8 informati
n
[RaiPlay] 027f044e-836d-4a3b-b1da-8eaa81715e3d: Downloading XML metadata f
r platform flash
[RaiPlay] 027f044e-836d-4a3b-b1da-8eaa81715e3d: Downloading XML metadata f
r platform native
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 251
[download] Destination: Che Dio ci aiuti S6E6 - Obbligo o verità-027f044e-
36d-4a3b-b1da-8eaa81715e3d.mp4
[download] 100% of 875.83MiB in 19:06

3 of 3)
where is the audio (not downloaded for unknown reasons)??

**Comment by 414n** [comment]
I had this too on some raiplay streams in the past.
I worked around it by:
 1. letting it download the video-only stream
 2. downloading the audio-only track by using `-f hls-aac-Italiano` with the same URL
 3. remuxing the two together in mkv format (mp4 had delay issues) using ffmpeg

**Comment by avagraha** [comment]
Thank you for your reply and suggestions, much appreciated!



  ----- Original Message -----

**Comment by fturco** [comment]
I have no problems with the following commands:

```
youtube-dl "https://www.raiplay.it/video/2021/01/Che-Dio-ci-aiuti-S6E6-Deja-vu-383bca47-d8a1-49c0-876c-f20a2d56d4c1.html" -o video.mp4
youtube-dl "https://www.raiplay.it/video/2021/01/Che-Dio-ci-aiuti-S6E6-Deja-vu-383bca47-d8a1-49c0-876c-f20a2d56d4c1.html" -f hls-aac-Italiano -o audio.mp4
ffmpeg -i video.mp4 -i audio.mp4 -c copy che-dio-ci-aiuti-s06e06.mp4
```

**Comment by avagraha** [comment]
Yes, ok. But I dont understand why the app is unable to download both 
streams (with merging process too) with the usual command, like in past.
Ex. gr. Pasty is able to donwload the complete mp4 (video+audio).



  ----- Original Message -----

**Comment by pjammo** [comment]
In this case `hls-1200`, `hls-1800` and `hls-2400` have no audio, they should be marked as `video only` formats.
Temporary workaround: `-f best+bestaudio`.

**Comment by christian-0s** [comment]
> In this case `hls-1200`, `hls-1800` and `hls-2400` have no audio, they should be marked as `video only` formats.
> Temporary workaround: `-f best+bestaudio`.

Unfortunately this doesn't work for me, audio is ok only for first 7 minutes, then it's muted.

**Comment by christian-0s** [comment]
Seems that https://github.com/ytdl-org/youtube-dl/pull/27990 fixed this issue

**Comment by fabzzap** [comment]
Found more examples of this. Could the suggested fix by @audetto be implemented?

**Comment by thePanz** [comment]
The solution from @audetto works for me.

Sidenote: the download speed of the RaiPlay videos/streams are *way* faster when ffmpeg is used, instead of the native method.

Try with: `-f 'hls-2400+hls-aac-Italiano' --hls-prefer-ffmpeg` (download went from 10mins to 2mins for a ~50min episode)
